src/main/java/com/amazonaws/services/kinesis/leases/impl/KinesisClientLease.java

```diff
@@ -142,6 +153,15 @@ public void setParentShardIds(Collection<String> parentShardIds) {
         this.parentShardIds.clear();
         this.parentShardIds.addAll(parentShardIds);
     }
+
+    /**
+     * Sets childShardIds.
+     *
+     * @param childShardIds may not be null
+     */
+    public void setChildShardIds(Collection<String> childShardIds) {
+        this.childShardIds.addAll(childShardIds);
```

Comment:
Either rename this addChildShardIds, or use `childShardIds`, this will not only set if this is called more than once.

Reply:
This methods should only be triggered once for one lease. It's immutable. So it should be fine if we call it setChildShardIds